sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -1042,15 +1078,13 @@ private static Cluster getCluster(
 
     private final Cluster cluster;
     private final Session session;
-    private final MappingManager mappingManager;
-    private List<ListenableFuture<Void>> mutateFutures;
-    private final BiFunction<Mapper<T>, T, ListenableFuture<Void>> mutator;
+    private final MapperFactory<T> mapperFactory;
+    private List<Future<Void>> mutateFutures;
+    private final BiFunction<Mapper<T>, T, Future<Void>> mutator;
     private final String operationName;
+    private final Class<T> entitiyClass;
 
-    Mutator(
-        Write<T> spec,
-        BiFunction<Mapper<T>, T, ListenableFuture<Void>> mutator,
-        String operationName) {
+    Mutator(Write<T> spec, BiFunction<Mapper<T>, T, Future<Void>> mutator, String operationName) {
```

Comment:
Ah, good question!
I had to do this because of a catch-22. We enforce vendored guava and as such the `Mapper` would have to use the vendored guava in the method signature. The Cassandra mapper `deleteAsync` and  `saveAsync` however returns the non-vendored guava so that left me with two options:

1. Keep vendored guava signature but deep copy/wrap the non-vendored `ListenableFuture` to a vendored `ListenableFuture` inside the `DefaultObjectMapper`.
2. Let the `Mapper` simply return `java.util.concurrent.Future` (which `ListenableFuture` extends anyway).

I went with option 2.

Reply:
Nice, this improvement is already worth the PR (IMO)!